metrics-operator/controllers/metrics/controller.go

```diff
@@ -142,3 +170,111 @@ func (r *KeptnMetricReconciler) fetchProvider(ctx context.Context, namespacedMet
 	}
 	return provider, nil
 }
+
```

Comment:
These aggregation functions might be useful in other places as well - I would suggest to move them to a separate package in `controllers/common/aggregation` and make them exported